src/analyses/base.ml

```diff
@@ -2274,6 +2276,8 @@ struct
       let return_val = project_val (Analyses.ask_of_ctx ctx) (attributes_varinfo (return_varinfo ()) callerFundec) (Some p) return_val (is_privglob (return_varinfo ())) in
       let cpa' = project (Analyses.ask_of_ctx ctx) (Some p) nst.cpa callerFundec in
 
+      if get_bool "sem.noreturn.dead_code" && Cil.hasAttribute "noreturn" f.svar.vattr then raise Deadcode;
```

Comment:
This applies to normal defined functions, but something similar should be in `special` I suppose. There could be a function which is declared to be noreturn, but doesn't provide an implementation.